auth/src/api/builder.rs

```diff
@@ -0,0 +1,79 @@
+use std::net::SocketAddr;
+
+use axum::{
+    middleware::from_extractor,
+    routing::{get, post},
+    Router, Server,
+};
+use shuttle_common::{
+    backends::metrics::{Metrics, TraceLayer},
+    request_span,
+};
+use sqlx::SqlitePool;
+use tracing::field;
+
+use crate::user::UserManager;
+
+use super::handlers::{
+    convert_cookie, convert_key, get_public_key, get_user, login, logout, post_user, refresh_token,
+};
+
+#[derive(Clone)]
+pub struct RouterState {
+    pub user_manager: UserManager,
+}
+
+pub struct ApiBuilder {
+    router: Router<RouterState>,
+    pool: Option<SqlitePool>,
+}
+
+impl Default for ApiBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ApiBuilder {
+    pub fn new() -> Self {
+        let router = Router::new()
+            .route("/login", post(login))
+            .route("/logout", post(logout))
+            .route("/auth/session", post(convert_cookie))
+            .route("/auth/key", post(convert_key))
+            .route("/auth/refresh", post(refresh_token))
+            .route("/public-key", get(get_public_key))
+            .route("/user/:account_name", get(get_user))
+            .route("/user/:account_name/:account_tier", post(post_user))
+            .route_layer(from_extractor::<Metrics>())
+            .layer(
+                TraceLayer::new(|request| {
+                    request_span!(request, request.params.account_name = field::Empty)
```

Comment:
We should add an `, account.name = field::Empty` for the guard to record correctly.

Reply:
I assume you meant add `account_tier`, which I did :smile: